### PR TITLE
sdk/node: new SDK

### PR DESCRIPTION
**Pull request by dominic**

### CODE REVIEW

**Comment by dominic** [comment]
Still WIP, and sorting out this CLA issue.

**Comment by jeffomatic** [comment]
In the example files, I notice that we use both `(foo) =>` and `function (foo)` style for lambdas. I don't think it makes a difference, so it'd be nice to be consistent. I think I prefer `(foo) =>`.

File: sdk/node/test/callbacks.js [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+        assert.equal(batchResponse.successes[0], null)
+        assert.deepEqual([batchResponse.errors[1], batchResponse.errors[2]], [null, null])
+        next()
+      }),
+
+
+      // Basic issuance
+
+      (next) => async.waterfall([
+        cb => client.transactions.build(function(builder){
```
Comment (jeffomatic) [inline sdk/node/test/callbacks.js]: Can we do `build(builder => ...`?

Comment (dominic) [inline sdk/node/test/callbacks.js]: The promises test will have the same problem.

File: sdk/node/src/connection.js [hunk 2]
```diff
@@ -5,6 +5,53 @@ const { fetch } = require('fetch-ponyfill')()
 const errors = require('./errors')
 const btoa = require('btoa')
 
+const blacklistAttributes = [
+  'asset_tags',
+  'asset_definition',
+  'account_tags',
+  'reference_data',
```
Comment (jeffomatic) [inline sdk/node/src/connection.js]: Should probably add:

- `tags`: used in assets.create, accounts.create
- `next`, `after`: pagination cursor in response and request, respectively

We could probably refer to the Swagger spec for any more instances of arbitrary objects.

File: sdk/node/src/connection.js [hunk 3]
```diff
@@ -34,6 +81,8 @@ class Connection {
       body = {}
     }
 
+    const snakeBody = snakeize(body) // Ssssssssssss
```
Comment (jeffomatic) [inline sdk/node/src/connection.js]: Love this comment, but I would like to expand on it to explain why this is happening.

File: sdk/node/src/connection.js [hunk 4]
```diff
@@ -111,6 +160,8 @@ class Connection {
             requestId: requestId
           }
         )
+      }).then((body) => {
+        return camelize(body)
```
Comment (jeffomatic) [inline sdk/node/src/connection.js]: Comment here to indicate why this is happening.

File: sdk/node/testSrc/callbacks.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ const balanceByAssetAlias = (cb) => {
     let res = {}
 
     balances.items.forEach((item) => {
-      res[item.sum_by['asset_alias']] = item.amount
+      res[item.sumBy['assetAlias']] = item.amount
```
Comment (jeffomatic) [inline sdk/node/testSrc/callbacks.js]: I realize this is old code, but it's probably ok to do `item.sumBy.assetAlias` here.

File: sdk/node/testSrc/callbacks.js [hunk 6]
```diff
@@ -413,7 +413,7 @@ describe('Callback style', () => {
         next()
       }),
 
-      cb => done()
+      () => done()
```
Comment (jeffomatic) [inline sdk/node/testSrc/callbacks.js]: you can just pass `done` here without the lambda

File: sdk/node/testSrc/promises.js [hunk 7]
```diff
@@ -13,7 +13,7 @@ const balanceByAssetAlias = (balances) => {
   return Promise.resolve(balances)
   .then((balance) => {
     balance.items.forEach((item) => {
-      res[item.sum_by['asset_alias']] = item.amount
+      res[item.sumBy['assetAlias']] = item.amount
```
Comment (jeffomatic) [inline sdk/node/testSrc/promises.js]: `item.sumBy.assetAlias`

**Comment by dominic** [comment]
PTAL

**Pull request merged**